### PR TITLE
Allow user to enable and disable hash/group aggregates with GPORCA

### PR DESCRIPTION
**Pull request by vraghavan78**
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by vraghavan78** [comment]
Not going into GPDB5

File: src/backend/utils/misc/guc_gp.c [hunk 1]
```diff
@@ -2382,6 +2384,28 @@ struct config_bool ConfigureNamesBool_gp[] =
 		NULL, NULL, NULL
 	},
 
+	{
+		{"optimizer_enable_hashagg", PGC_USERSET, DEVELOPER_OPTIONS,
+			gettext_noop("Enables Pivotal Optimizer (GPORCA) to use of hashed aggregation plans."),
```
Comment (bhuvnesh2703) [inline src/backend/utils/misc/guc_gp.c]: `to use of hashed aggregation` -> `to use hash aggregation plan` or `to use hash aggregates`

File: src/backend/utils/misc/guc_gp.c [hunk 2]
```diff
@@ -2382,6 +2384,28 @@ struct config_bool ConfigureNamesBool_gp[] =
...
+			GUC_NO_SHOW_ALL | GUC_NOT_IN_SAMPLE
+		},
+		&optimizer_enable_hashagg,
+		true,
+		NULL, NULL, NULL
+	},
+
+	{
+		{"optimizer_enable_groupagg", PGC_USERSET, DEVELOPER_OPTIONS,
+			gettext_noop("Enables Pivotal Optimizer (GPORCA) to use of hashed aggregation plans."),
```
Comment (bhuvnesh2703) [inline src/backend/utils/misc/guc_gp.c]: similar as above comment.

File: src/test/regress/expected/qp_orca_fallback_optimizer.out [hunk 3]
```diff
@@ -215,3 +215,45 @@ DETAIL:  Feature not supported: ONLY in the FROM clause
...
+               Group Key: a
+               ->  Sort  (cost=0.00..431.98 rows=3334 width=4)
+                     Sort Key: a
+                     ->  Seq Scan on foo  (cost=0.00..431.07 rows=3334 width=4)
+ Optimizer: Pivotal Optimizer (GPORCA) version 3.39.0
+(8 rows)
+
+set optimizer_enable_groupagg = off;
+explain select count(*) from foo group by a;
+INFO:  GPORCA failed to produce a plan, falling back to planner
```
Comment (bhuvnesh2703) [inline src/test/regress/expected/qp_orca_fallback_optimizer.out]: could we have it this way (its trivial though but i think is more obvious)
1. Both off -> fallback
2. Group agg on -> get group agg plan
3. Group agg off and Hash Agg on -> get hash agg plan

File: src/backend/utils/misc/guc_gp.c (hunk 2 above)
Comment (hardikar) [reply src/backend/utils/misc/guc_gp.c]: Except "group aggregate" plans

File: src/test/regress/expected/qp_orca_fallback_optimizer.out [hunk 4]
```diff
@@ -215,3 +215,45 @@ DETAIL:  Feature not supported: ONLY in the FROM clause
  Optimizer: Postgres query optimizer
 (5 rows)
 
+create table foo(a int, b int);
+NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
+HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
+insert into foo select i%100, i%100 from generate_series(1,10000)i;
+explain select count(*) from foo group by a;
```
Comment (hardikar) [inline src/test/regress/expected/qp_orca_fallback_optimizer.out]: Can you explicitly enable the 2 GUCs before this?

File: src/test/regress/expected/qp_orca_fallback_optimizer.out [hunk 5]
```diff
@@ -215,3 +215,45 @@ DETAIL:  Feature not supported: ONLY in the FROM clause
...
+--------------------------------------------------------------------------------
+ Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..431.50 rows=100 width=8)
+   ->  Result  (cost=0.00..431.50 rows=34 width=8)
+         ->  HashAggregate  (cost=0.00..431.50 rows=34 width=8)
+               Group Key: a
+               ->  Seq Scan on foo  (cost=0.00..431.07 rows=3334 width=4)
+ Optimizer: Pivotal Optimizer (GPORCA) version 3.39.0
+(6 rows)
+
+set optimizer_enable_hashagg = off;
```
Comment (hardikar) [inline src/test/regress/expected/qp_orca_fallback_optimizer.out]:
```suggestion
set optimizer_enable_hashagg = off;
set optimizer_enable_groupagg = on;
```

File: src/test/regress/expected/qp_orca_fallback_optimizer.out [hunk 6]
```diff
@@ -215,3 +215,45 @@ DETAIL:  Feature not supported: ONLY in the FROM clause
...
+   ->  Result  (cost=0.00..431.99 rows=34 width=8)
+         ->  GroupAggregate  (cost=0.00..431.99 rows=34 width=8)
+               Group Key: a
+               ->  Sort  (cost=0.00..431.98 rows=3334 width=4)
+                     Sort Key: a
+                     ->  Seq Scan on foo  (cost=0.00..431.07 rows=3334 width=4)
+ Optimizer: Pivotal Optimizer (GPORCA) version 3.39.0
+(8 rows)
+
+set optimizer_enable_groupagg = off;
```
Comment (hardikar) [inline src/test/regress/expected/qp_orca_fallback_optimizer.out]:
```suggestion
set optimizer_enable_hashagg = off;
set optimizer_enable_groupagg = off;
```

**Pull request merged**